.github/workflows/ci.yml

```diff
@@ -4,42 +4,34 @@ on: [push]
 
 env:
   CI: true
-  DEFAULT_NODE_VERSION: 16.x
 jobs:
   build:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+          cache: "npm"
+          cache-dependency-path: "**/package-lock.json"
+      - run: npm ci --ignore-scripts
+      - run: npm run build
+  test:
+    needs: build
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         # Available OS's: https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners
         os: [ubuntu-20.04, windows-2019, macos-10.15]
         node-version: [16.x, 14.x, 12.x]
     steps:
-      - uses: actions/checkout@v2.4.0
+      - uses: actions/checkout@v2
       - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v2.5.1
+        uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node-version }}
-      - name: Cache node modules
-        uses: actions/cache@v2.1.7
-        env:
-          cache-name: cache-node-modules
-        with:
-          path: node_modules
-          key: ${{ runner.os }}-node${{ runner.node-version }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
-      # Retry `npm ci` if it fails (No, GitHub Actions at this point can retry)
-      - run: npm ci || npm ci || npm ci
-      - run: npm run build
+          cache: "npm"
+          cache-dependency-path: "**/package-lock.json"
+      - run: npm ci --ignore-scripts
       - run: npm test
-      - run: npx package-check
-      - name: Archive code coverage results
```

Comment:
We don't actually generate a code coverage artifact as a file anymore, as of #1386 — instead the coverage is just reported to the console output.